nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/Lock.java

```diff
@@ -77,4 +80,36 @@ public boolean getFailed() {
         return failed;
     }
 
+    /**
+     * @since 11.1
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if(obj instanceof Lock) {
+            Lock other = (Lock) obj;
+            boolean result = false;
+            result = this.owner.equals(other.owner);
+            result = result && this.created.equals(other.created);
+            result = result && this.failed == other.failed;
+            return result;
+        }
+        return false;
+    }
+
+    /**
+     * @since 11.1
+     */
+    @Override
+    public int hashCode() {
+        return cachedHashCode;
+    }
+
+    private int calculateHashCode() {
+        final int prime = 31;
```

Comment:
Same thing than `EqualsBuilder` but for `HashCodeBuilder` here.

Note: There's also `ToStringBuilder` from the same library.

Reply:
Thanks for the tips !